order/src/main/java/com/yas/order/service/CartItemService.java

```diff
@@ -48,10 +64,14 @@ private void updateExistingCartItem(CartItemPostVm cartItemPostVm, CartItem exis
     }
 
     private void validateCartItemPostVm(CartItemPostVm cartItemPostVm) {
+        validateProduct(cartItemPostVm.productId());
+    }
+
+    private void validateProduct(Long productId) {
         try {
-            productService.getProductById(cartItemPostVm.productId());
+            productService.getProductById(productId);
         } catch (NotFoundException e) {
-            throw new BadRequestException(Constants.ErrorCode.PRODUCT_NOT_FOUND, cartItemPostVm.productId());
+            throw new BadRequestException(Constants.ErrorCode.PRODUCT_NOT_FOUND, productId);
         }
     }
```

Comment:
Consider combining to one private function

Reply:
Hi @minhtranq-nashtechglobal, thank you for your feedback earlier. I've just pushed the changes. Could you please take a look when you have time?